src/main/java/org/opensearch/knn/engine/method/EngineSpecificMethodContext.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.engine.method;
+
+import org.opensearch.knn.index.Parameter;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Holds context related to a method for a particular engine
+ * Each engine can have a specific set of parameters that it supports during index and build time. This context holds
+ * the information for each engine method combination.
+ *
+ * TODO: Move KnnMethod in here
+ */
+public interface EngineSpecificMethodContext {
```

Comment:
What do you think about name `MethodImplementationContext`?

Reply:
I really like having "engine/nativelibrary" and "method" both in there to clearly indicate that it shouldn't be used stand alone. Any other suggestions?

Reply:
> I really like having "engine/nativelibrary" and "method" both in there to clearly indicate that it shouldn't be used stand alone ideally

What do you mean by stand alone here?

Reply:
For instance, I can see someone using it to use this to get HNSW method parameters in universal fashion and plug it somewhere other than engine.